src/serviceDialog.js

```diff
@@ -87,6 +151,62 @@
     });
     grid.attach(URLEntry, 2, 2, 1, 1);
 
+    const iconLabel = new Label({
+      label: "Icon",
+      halign: Align.END,
+    });
+    grid.attach(iconLabel, 1, 3, 1, 1);
+    const iconEntry = new Entry({
+      text: showIcon,
+      hexpand: true,
+    });
+    grid.attach(iconEntry, 2, 3, 1, 1);
+
+    const fileButton = new Gtk.Button({ label: "choose" });
+    grid.attach(fileButton, 3, 3, 1, 1);
+
+    // Bind it to a function that says what to do when the button is clicked
+    fileButton.connect("clicked", () => {
+      const file = openIconChooserDialog();
+      if (file) {
+        iconEntry.text = file;
+      }
+    });
```

Comment:
Could you use FileChooserButton instead? No need to reinvent the wheel, FileChooserButton does exactly this.